.github/actions/cancel-workflow/action.yml

```diff
@@ -0,0 +1,16 @@
+name: Cancel Workflow
+description: >
+  Cancels the current workflow run, i.e. all jobs. Useful if you want to cancel the rest of the workflow when one job
+  fails.
+
+runs:
+  using: "composite"
+  steps:
+    - name: Cancel Workflow
+      # Fork PRs won't have a token with write access to Actions, thus won't be able to cancel the workflow.
+      if: github.event.pull_request == '' || github.event.pull_request.head.repo.fork == false
```

Comment:
We use this condition in `check-merge-conflict` workflow too. Can't we DRY this?
I mean we could add a `setup` step and with `Set-Output` we can add eg.: `has-write-access` (until no official solution for this) and check the `has-write-access` here and in `check-merge-conflict`.

What do you think about this?

Reply:
I'd love to DRY this but we'd have the opposite effect. For this, we'd need a composite action and call it like this:

```yml
- name: Check Fork PR
  id: check-fork-pr
  uses: Lombiq/GitHub-Actions/.github/actions/check-fork-pr@dev

- name: Cancel Workflow
  if: steps.check-fork-pr.outputs.is-fork-pr != 'true'
...
```

This duplicates the amount of code we'd need to copy-paste around (plus the action itself). While granted, the logic and docs, which matter more, would be centralized (to an extent; the docs are slightly different due to the different operations), this still seems like a bad deal to me.